Source/Modules/fortran.cxx

```diff
@@ -0,0 +1,4101 @@
+
+
+/* -------------------------------------------------------------------------
+ * \brief Process callbacks, which generate 'getter' wrapper functions
+ *
+ * To avoid breaking the later 'functionWrapper', we create a copy of the node.
+ */
+int FORTRAN::callbackfunctionHandler(Node *n) {
+  // Create a shallow copy of the node with params
+  Node *cbnode = copyNode(n);
+  Setattr(cbnode, "parms", Getattr(n, "parms"));
+  Language::callbackfunctionHandler(cbnode);
+  Setattr(n, "feature:callback:node", cbnode);
+  return SWIG_OK;
+}
+
+/* -------------------------------------------------------------------------
+ * \brief Process *compile-time* constants
+ *
+ * These include:
+ * \code
+    %callback("%s_cb") add;
+    %constant int wrapped_const = (1 << 3) | 1;
+    #define MY_INT 0x123
+    constexpr int myint = 4;
+ * \endcode
+ * and enum values.
+ *
+ * - Native enum values will become enumerators
+ * - Non-native enum values become C-bound external constants
+ * - Constants marked with `%fortranconst` will be rendered as *named constants*
+ * - Constants marked with `%fortranbindc` also become C-bound external constants
+ * - All other types will generate `getter` functions that return native fortran types.
+ */
+int FORTRAN::constantWrapper(Node *n) {
+  enum {
+    NATIVE_ENUM,
+    EXTERN_ENUM,
+    NATIVE_CONSTANT,
+    EXTERN_CONSTANT
+  } constant_type;
+
+  if (d_enum_public) {
+    constant_type = NATIVE_ENUM;
+  } else if (Cmp(nodeType(n), "enumitem") == 0) {
+    constant_type = EXTERN_ENUM;
+  } else if (GetFlag(n, "feature:fortran:const") || Getattr(n, "feature:fortran:constvalue")) {
+    constant_type = NATIVE_CONSTANT;
+  } else if (GetFlag(n, "feature:fortran:bindc") && has_constant_storage(n)) {
+    constant_type = EXTERN_CONSTANT;
+  } else {
+    // Not a natively representable value: wrap as immutable global variable
+    SetFlag(n, "feature:immutable");
+    return this->globalvariableHandler(n);
+  }
+
+  // Get symbolic name
+  String *fsymname = this->get_fsymname(n);
+  if (!fsymname) {
+    return SWIG_ERROR;
+  }
+
+  if (constant_type == NATIVE_ENUM) {
+    // Add to list of public enums
+    Append(d_enum_public, fsymname);
+    Printv(f_fdecl, "  enumerator :: ", fsymname, NULL);
+    if (String *value = Getattr(n, "enumvalue")) {
+      Printv(f_fdecl, " = ", value, NULL);
+    }
+    Printv(f_fdecl, "\n", NULL);
+    return SWIG_OK;
+  }
+
+  // Attach bindc typemaps
+  String *bindc_typestr = attach_typemap("bindc", n, WARN_NONE);
+  if (!bindc_typestr) {
+    Swig_error(input_file,
+               line_number,
+               "The constant '%s' has no 'bindc' typemap.\n",
+               Getattr(n, "sym:name"));
+    return SWIG_ERROR;
+  }
+
+  // Get value of the constant
+  String *value = (constant_type == NATIVE_CONSTANT ? Getattr(n, "feature:fortran:constvalue") : NULL);
+  if (!value) {
+    // String types will have the quoted, escaped value set as rawval
+    value = Getattr(n, "rawval");
+  }
```

Comment:
Once #2990 is merged, this will need updating.

The `rawval` attribute is gone, and `value` will be a valid C/C++ expression (so for a string or char literal it's suitably escaped and includes quotes).

So I think the `if` above can just be removed.